src/pickers/UniqueCodeInput.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+import clsx from "clsx";
+import { TextInput, useDebounceCb, useModulesManager, useTranslations, useGraphqlQuery } from "@openimis/fe-core";
+import { InputAdornment, CircularProgress, Box } from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+import CheckOutlinedIcon from "@material-ui/icons/CheckOutlined";
+import ErrorOutlineOutlinedIcon from "@material-ui/icons/ErrorOutlineOutlined";
+
+const useStyles = makeStyles((theme) => ({
+  validIcon: {
+    color: "green",
+  },
+  invalidIcon: {
+    color: theme.palette.error.main,
+  },
+}));
+
+const operation = `
+  query ($locationCode: String!) {
+    isValid: validateLocationCode(locationCode: $locationCode)
+  }
+`;
+
+const UniqueCodeInput = (props) => {
+  const { value, new_location, onChange, className, label = "Code", placeholder, readOnly, required, inputProps } = props;
+  const [internalValue, setInternalValue] = useState(value);
+  const modulesManager = useModulesManager();
+  const { formatMessage } = useTranslations("location", modulesManager);
+  const classes = useStyles();
+  const {
+    isLoading,
+    data,
+    error: graphqlError,
+  } = useGraphqlQuery(operation, { locationCode: internalValue}, { skip: !internalValue });
+
+  const handleValueChange = useDebounceCb((val) => {
+    if (val) {
+      setInternalValue(val);
+    } else {
+      onChange({"code": val, "isValid": isValid});
+    }
+  }, modulesManager.getConf("fe-location", "debounceTime", 400));
+
+
+  const isValid = !isLoading && data?.isValid;
+  const isInvalid = !isLoading && data && !data.isValid;
```

Comment:
We talked about it before. What's the point of declaration isValid and isInvalid variables? Maybe we should just remove this isInvalid and use !isValid instead.

Reply:
I just did, thank you for the suggestion. I will mark all convos regarding this as resolved.